hot-reload.md

```diff
@@ -0,0 +1,279 @@
+
+Conversely, in the following example:
+```
+const foo = 1;
+final bar = foo;
+print(foo);
+print(bar);
+```
+
+
+running the app for the first time prints `1` and `1`. Then if you make the
+following change: 
+
+```
+const foo = 2;    //modified
+final bar = foo;
+print(foo);
+print(bar);
+```
+and hot reload, it now prints `2` and `1`. While changes to `const` field values
+are always hot reloaded, the static field initializer is not rerun.
+Conceptually, `const` fields are treated like aliases instead of state.
+
+The Dart VM detects initializer changes and flags when a set of changes needs a
+full restart to take effect. The flagging mechanism is triggered for most of the
+initialization work in the above example, but not for cases like:
+
+```
+final bar = foo;
+```
+
+To be able to update `foo` and view the change after hot reload, consider
+redefining the field as `const` or using a getter to return the value, rather
+than using `final`. For example:
+
+```
+const bar = foo;
+```
+or:
+
+```
+get bar => foo;
+```
+
+Read more about the [differences between the `const` and `final` keywords](https://news.dartlang.org/2012/06/const-static-final-oh-my.html) in Dart. 
+
+
+### Previous state is combined with new code 
+
+If code changes affect the state of your app (or its dependencies), the Flutter
+framework gives you two options for reconciling old data with new code: throw
+away state or modify the existing state. Throwing away state is what happens
+during a full restart. Depending on the situation, you might prefer to apply
+changes in a more selective way, by marking the widget tree as dirty and
+recomputing the derived layout trees, while leaving the UI state alone. 
+
+Modifying the existing state is what happens during hot reload; however, in this
+scenario the data your app has to work with is not fully consistent with the
+data it would have if it executed from scratch. Thus your app might behave
+differently after hot reload versus a full restart.
+
+For example, if you modify a class definition from extending StatelessWidget to
+StatefulWidget (or the reverse), after hot reload the previous state of your
+app is preserved. However, the state might not be compatible with the new
+changes.
+
+Consider the following code:
+
+```
+class myWidget extends StatelessWidget {
+  Widget build(BuildContext context) {
+    return new GestureDetector(onTap: () => print('T'));
+  }
+}
+```
+After running the app, if you make the following change:
+
+```
+class myWidget extends StatefulWidget {
+  @override
+  State createState() => new myWidgetState();
+}
+class myWidgetState {
+...
+...
+}
+```
+
+and then hot reload, the console displays an assertion failure similar to:
+
+```
+myWidget is not a subtype of StatelessWidget
+```
+
+## Understanding limitations
+
+In addition to edge cases, you might also encounter the rare scenarios where hot
+reload is not supported at all.  These include:
+
+*   Changed source files have compilation errors
```

Comment:
This really isn't a "limitation", it's a feature

Reply:
Users will likely see it as a limitation, and not in a negative way...it can be very good to have something feature-like first on the list.

If you can think of a sentence to add that gives some reassuring context here, I'll add it.